lib/hal_api/controller/exceptions.rb

```diff
@@ -7,22 +7,27 @@ module HalApi::Controller::Exceptions
   extend ActiveSupport::Concern
 
   def respond_with_error(exception)
-    wrapper = if HalApi.rails_major_version >= 5
-                ::ActionDispatch::ExceptionWrapper.new(ActiveSupport::BacktraceCleaner.new, exception)
-              else
-                ::ActionDispatch::ExceptionWrapper.new(request.env, exception)
-              end
-
-    log_error(request.env, wrapper)
-
-    error = exception
-    if !error.is_a?(HalApi::Errors::ApiError)
-      error = HalApi::Errors::ApiError.new(error.message).tap do |e|
-        e.set_backtrace(error.backtrace)
+    wrapper =
+      if HalApi.rails_major_version >= 5
+        ::ActionDispatch::ExceptionWrapper.new(ActiveSupport::BacktraceCleaner.new, exception)
+      else
+        ::ActionDispatch::ExceptionWrapper.new(request.env, exception)
       end
-    end
 
-    notice_error(exception) if error.status >= 500
+    error =
+      if exception.is_a?(HalApi::Errors::ApiError) && exception.status >= 500
+        log_error(request.env, wrapper)
+        notice_error(exception)
+        exception
+      elsif exception.is_a?(HalApi::Errors::ApiError)
```

Comment:
So this is a non-5xx error, like a 404 or 401 - those won't get logged at all, not even to info or debug?

Reply:
I think I would still log these just not as `fatal`, maybe pass in a severity to the `log_error` method?

Reply:
Yeah, I thought that a first too.  But turns out their statuses get logged by the regular rails request logger - so you see a response going out with status=401 (or whatever code).  And in the cases I checked, there was no additional info gained by `logger.warn`-ing the error itself.